gui/system/forms.py

```diff
@@ -2705,6 +2717,19 @@ class Meta:
         model = models.Certificate
 
 
+def is_valid_privatekey(certificate, privatekey):
+    pkey_obj = crypto.load_privatekey(crypto.FILETYPE_PEM, privatekey)
+    cert_obj = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
+    ctx = SSL.Context(SSL.TLSv1_METHOD)
```

Comment:
I tested it out for CAs at my end when you asked for it. It works, so I can add it for CAs too.
Also while testing I realized that the code breaks for RSA keys.